File: android/src/app/src/main/java/com/example/agitodo/Logger.java
package com.example.agitodo;

import android.util.Log;

public class Logger {
    static boolean debug = false;

    public void info(String msg) {
        if (debug) {
            Log.i("Agitodo", msg);
        }
    }

    public void error(String msg) {
        if (debug) {
            Log.e("Agitodo", msg);
        }
    }
}

File: www/logger.js
var path = require("path");
var fs = require("fs");

var logFile = path.join(__dirname, "log/agitodo.log");

function log(level, msg) {
  var fields = [];
  fields.push("[" + new Date().toISOString() + "]");
  fields.push(level);
  fields.push(typeof msg === "string" ? msg : JSON.stringify(msg));

  // 432 --> 0660 (octal)
  fs.appendFile(logFile, fields.join(" ") + "\n", {
    "mode": 432
  }, function(err) {
    if (err) {
      console.log(err);
    }
  });
}

module.exports = {
  info: function(msg) {
    log("INFO", msg);
  },
  warning: function(msg) {
    log("WARNING", msg);
  },
  error: function(msg) {
    log("ERROR", msg);
  }
};
